src/hdmf/data_utils.py

```diff
@@ -626,6 +626,8 @@ def __bool__(self):
 
     def __getattr__(self, attr):
         """Delegate attribute lookup to data object"""
+        if attr == '__array_struct__':  # np.array() checks __array__ or __array_struct__ attribute dep. on version
```

Comment:
Does this mean delegation of ``__array__`` is allows but delegation of ``__array_struct__`` is not? I.e., depending on the version of numpy we allow conversion of DataIO to a numpy array or not? Could you clarify what the problem with the ``__array_struct__`` method is?

Reply:
My mistake. The error should only be thrown if the object is not self.valid. I have added that back.